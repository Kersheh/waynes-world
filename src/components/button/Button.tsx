import React from 'react';
import classNames from 'classnames/bind';

import styles from './Button.module.scss';

const cx = classNames.bind(styles);

export interface ButtonProps {
  children?: React.ReactNode;
  name?: string;
  type?: 'button' | 'reset' | 'submit';
  disabled?: boolean;
  styleType?: 'basic' | 'text' | 'fullWidth' | 'tile' | 'listItem';
  onClick?: (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void;
}
const Button = ({
  children,
  name,
  type = 'button',
  disabled = false,
  styleType = 'text',
  onClick
}: ButtonProps) => {
  return (
    <button
      name={name}
      type={type}
      disabled={disabled}
      onClick={onClick}
      className={cx({
        button: true,
        basic: styleType === 'basic',
        text: styleType === 'text',
        fullWidth: styleType === 'fullWidth',
        tile: styleType === 'tile',
        listItem: styleType === 'listItem'
      })}
      onTouchStart={() => {}}
    >
      {children}
    </button>
  );
};

export default Button;
