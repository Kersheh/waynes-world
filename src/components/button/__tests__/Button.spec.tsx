import React from 'react';
import { render } from '@testing-library/react';

import Button from '../Button';

describe('<Button/> component', () => {
  it('should render', () => {
    const { container } = render(<Button />);

    expect(container).toMatchSnapshot();
  });
});
