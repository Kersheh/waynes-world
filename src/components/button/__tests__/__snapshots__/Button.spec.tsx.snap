// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button/> component should render 1`] = `
<div>
  <button
    class="button text"
    type="button"
  />
</div>
`;
